client/app/components/routes/profile/User.tsx

```diff
@@ -28,7 +28,13 @@ export default async function User() {
        * getServerSession requires passing the same object you would pass to NextAuth
        */
       const session = await getServerSession(authOptions);
-      const names = session?.user?.name?.split(" ");
+      const isIdir = session?.identity_provider === "idir";
+      // IDIR names come in the format "LASTNAME, FIRSTNAME" so we will split on the comma
+      // and reverse so they don't go into the wrong fields
+      const idirName = session?.user?.name?.split(", ").reverse();
```

Comment:
My IDIR includes my middle initial and ENV:INXT apparently: 
![image](https://github.com/bcgov/cas-registration/assets/77306764/da61d836-1c6d-428c-8df5-b6af04d4524f)
But I think that's ~probably something to do with the way mine is set up specifically

Reply:
Mine does as well, though mine is `ENV:EX` though I decided it was out of scope for this.

Reply:
I think the point of this ticket was to stop bad data from going into the db, so if this is out of scope for this ticket, I think we need to check with Dylan about a second ticket. I'll make a placeholder tech debt one so we don't forget: https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/447